src/Camelot.Services.Windows/WinApi/ShellLink.cs

```diff
@@ -0,0 +1,191 @@
+        SLR_UPDATE = 0x4,
+        /// <summary>Do not update the link information</summary>
+        SLR_NOUPDATE = 0x8,
+        /// <summary>Do not execute the search heuristics</summary>
+        SLR_NOSEARCH = 0x10,
+        /// <summary>Do not use distributed link tracking</summary>
+        SLR_NOTRACK = 0x20,
+        /// <summary>Disable distributed link tracking. By default, distributed link tracking tracks
+        /// removable media across multiple devices based on the volume name. It also uses the
+        /// Universal Naming Convention (UNC) path to track remote file systems whose drive letter
+        /// has changed. Setting SLR_NOLINKINFO disables both types of tracking.</summary>
+        SLR_NOLINKINFO = 0x40,
+        /// <summary>Call the Microsoft Windows Installer</summary>
+        SLR_INVOKE_MSI = 0x80
+    }
+
+
+    /// <summary>The IShellLink interface allows Shell links to be created, modified, and resolved</summary>
+    [ComImport(), InterfaceType(ComInterfaceType.InterfaceIsIUnknown), Guid("000214F9-0000-0000-C000-000000000046")]
+    interface IShellLinkW
+    {
+        /// <summary>Retrieves the path and file name of a Shell link object</summary>
+        void GetPath([Out(), MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszFile, int cchMaxPath, out WIN32_FIND_DATAW pfd, SLGP_FLAGS fFlags);
+        /// <summary>Retrieves the list of item identifiers for a Shell link object</summary>
+        void GetIDList(out IntPtr ppidl);
+        /// <summary>Sets the pointer to an item identifier list (PIDL) for a Shell link object.</summary>
+        void SetIDList(IntPtr pidl);
+        /// <summary>Retrieves the description string for a Shell link object</summary>
+        void GetDescription([Out(), MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszName, int cchMaxName);
+        /// <summary>Sets the description for a Shell link object. The description can be any application-defined string</summary>
+        void SetDescription([MarshalAs(UnmanagedType.LPWStr)] string pszName);
+        /// <summary>Retrieves the name of the working directory for a Shell link object</summary>
+        void GetWorkingDirectory([Out(), MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszDir, int cchMaxPath);
+        /// <summary>Sets the name of the working directory for a Shell link object</summary>
+        void SetWorkingDirectory([MarshalAs(UnmanagedType.LPWStr)] string pszDir);
+        /// <summary>Retrieves the command-line arguments associated with a Shell link object</summary>
+        void GetArguments([Out(), MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszArgs, int cchMaxPath);
+        /// <summary>Sets the command-line arguments for a Shell link object</summary>
+        void SetArguments([MarshalAs(UnmanagedType.LPWStr)] string pszArgs);
+        /// <summary>Retrieves the hot key for a Shell link object</summary>
+        void GetHotkey(out short pwHotkey);
```

Comment:
Question: do we need all methods? We call like 1 or 2, can we remove the rest? Will it break anything?

Reply:
They are all part of the COM interface.
If I recall correctly the dotnet level is implement via IDispatch, and via ids (bcz tlb is not used here),
it shouldn't break anything. But it is "customary" to keep the whole methods when declaring it.

Reply:
If these methods can be safely removed let's do so, less code to support later